applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py

```diff
@@ -450,10 +460,23 @@ def _CreateSolutionStrategy(self):
         if analysis_type == "linear":
             mechanical_solution_strategy = self._create_linear_strategy()
         elif analysis_type == "non_linear":
-            if(self.settings["line_search"].GetBool() == False):
+            # Deprecation checks
+            if self.settings.Has("line_search"):
+                kratos_utilities.IssueDeprecationWarning('MechanicalSolver', 'Using "line_search", please move it to "solving_strategy_settings" as "strategy_type"')
+                if self.settings["line_search"].GetBool():
+                    self.settings["solving_strategy_settings"]["strategy_type"].SetString("line_search")
+            if self.settings.Has("arc_length"):
+                kratos_utilities.IssueDeprecationWarning('MechanicalSolver', 'Using "arc_length", please move it to "solving_strategy_settings" as "strategy_type"')
+                if self.settings["arc_length"].GetBool():
+                    self.settings["solving_strategy_settings"]["strategy_type"].SetString("arc_length")
```

Comment:
```suggestion
```

this does not exist yet hence no deprecation is necessary

Reply:
sure